index.bs

```diff
@@ -2028,9 +2039,235 @@ BrowserResult = (
+
+  1. [=list/Append=] |document| to |documents|.
+
+  1. If |document|'s [=visibility state=] is "<code>visible</code>",
+     [=list/Append=] |document| to |visible documents|.
+
+1. For each |document| in |visible documents|:
+
+  1. If |document|'s [=fullscreen element=] is not null, return
+     "<code>fullscreen</code>".
+
+1. If |visible documents| is [=list/empty=] but |documents| is not
+   [=list/empty=], or if |window| is otherwise in an OS-specific minimized
+   state, return "<code>minimized</code>".
+
+   Note: This will usually, but not necessarily, mean that |window|'s [=client
+   window/width=] and [=client window/height=] are equal to 0.
+
+1. If |window| is in an OS-specific maximized state return
+   "<code>maximized</code>".
+
+   Note: This will usually, but not necessarily, mean that |window|'s [=client
+   window/width=] is equal to the width of the [=web-exposed screen area=] and
+   |window|'s [=client window/height=] is equal to the height of the
+   [=web-exposed screen area=].
+
+1. Return "<code>normal</code>".
+
+</div>
+
+<div algorithm>
+
+To <dfn>set the client window state</dfn> given |window| and |state|:
+
+1. Let |current state| be [=get the client window state| with |window|.
+
+1. If |current state| is equal to |state|, return [=success=] with data null.
+
+1. Switch on the value of |state|:
+
+   <dl>
+    <dt>"<code>fullscreen</code>"
+    <dd>If not [=fullscreen is supported=] return [=error=] with [=error code=]
+    [=unsupported operation=].
+
+    <dt>"<code>normal</code>"
+    <dd>If not [=restore client window is supported=] for |window| return [=error=] with
+    [=error code=] [=unsupported operation=].
+
+    <dt>"<code>maximize</code>"
+    <dd>If not [=maximize client window is supported=] for |window| return [=error=] with
+    [=error code=] [=unsupported operation=].
+
+    <dt>"<code>minimize</code>"
+    <dd>If not [=minimize client window is supported=] for |window| return [=error=] with
+    [=error code=] [=unsupported operation=].
+   </dl>
+
+1. Let |documents| be an empty [=/list=].
+
+1. For each [=/top-level traversable=] |traversable|:
+
+  1. If |traversable|'s associated [=client window=] is not |window| then
+     continue.
+
+  1. Let |document| be |traversable|'s [=active document=].
+
+  1. Append |document| to |documents|.
+
+1. If |documents| is [=list/empty=] return [=error=] with [=error code=]
+   [=no such client window=].
+
+1. If |current state| is "<code>fullscreen</code>":
+
+  1. For each |document| in |documents|:
+
+    1. [=Fully exit fullscreen=] with |document|.
+
+       Note: This is a no-op for documents in window that are not fullscreen.
+
+1. Switch on the value of |state|:
+
+   <dl>
+    <dt>"<code>fullscreen</code>"
+    <dd>
+      1. For each |document| in |documents|:
+
+        1. If |document|'s [=visibility state=] is "<code>visible</code>",
+           [=fullscreen an element=] with |document|'s [=document element=].
+
+        1. [=Break=].
```

Comment:
Iterate through all the documents?
```suggestion
```

Reply:
That doesn't seem obvious? If you have more than one visible document in a specific window then you still can't really have more than one fullscreen element. At the moment we'd pick the first one (albeit in an arbitary order), with your proposed change we'd fullscreen an element from each document in turn. I don't quite see if anything in the fullscreen spec says that you have to unfullscreen any existing documents that are already fullscreen (maybe it's assumed that user activation requirements mean that you can't get into that situation?), but it at least feels confusing.